File: Backend/LinkCare/src/main/java/com/ssafy/linkcare/group/dto/JoinRequestResponse.java
package com.ssafy.linkcare.group.dto;

import com.ssafy.linkcare.group.enums.RequestStatus;

import java.time.LocalDateTime;

/*
    * 그룹 참가 신청 응답 DTO
    * 방장이 대기 중인 신청 목록을 볼 때 사용
*/
public record JoinRequestResponse(
        Long requestSeq,
        Long userSeq,
        String userName,
        Integer userAge,
        String userGender,
        RequestStatus status,
        LocalDateTime requestedAt
) {}

File: Frontend/app/src/main/java/com/a307/linkcare/feature/commongroup/data/model/response/JoinRequestResponse.kt
package com.a307.linkcare.feature.commongroup.data.model.response

import com.google.gson.annotations.SerializedName

data class JoinRequestResponse(
    @SerializedName("requestSeq")
    val requestSeq: Long,

    @SerializedName("userSeq")
    val userSeq: Long,

    @SerializedName("userName")
    val userName: String,

    @SerializedName("userAge")
    val userAge: Int,

    @SerializedName("userGender")
    val userGender: String,

    @SerializedName("status")
    val status: String, // PENDING, APPROVED, REJECTED

    @SerializedName("requestedAt")
    val requestedAt: String
)
